### ISSUE TITLE
the method getRequestHeaderMap in class  getRequestHeaderMap.HTTPRequestUtils

### ISSUE THREAD

**Issue by Kookey** [Netflix/zuul#393]
package:com.netflix.zuul.util
class : HTTPRequestUtils
method : getRequestHeaderMap

    public Map<String, List<String>> getRequestHeaderMap() {
        HttpServletRequest request = RequestContext.getCurrentContext().getRequest();
        Map<String,List<String>> headers = new HashMap<String,List<String>>();
        Enumeration<String> headerNames = request.getHeaderNames();
        if(headerNames != null) {
            while (headerNames.hasMoreElements()) {
                String name = headerNames.nextElement();
                String value = request.getHeader(name);

                if(name != null && !name.isEmpty() && value != null) {
                    List<String> valueList = new ArrayList<String>();
                    if(headers.containsKey(name)) {
                        headers.get(name).add(value);
                    }
                    valueList.add(value);
                    headers.put(name, valueList);
                }
            }
        }
        return Collections.unmodifiableMap(headers);

    }

this method have no test, but this method may can't work fine, because the code below 
List<String> valueList = new ArrayList<String>();
                    if(headers.containsKey(name)) {
                        headers.get(name).add(value);
                    }
                    valueList.add(value);
                    headers.put(name, valueList);
                }
can only return a list with one item,

if(headers.containsKey(name)) {
                        headers.get(name).add(value);
                    }
headers.get(name) return a list that in the map  and then add new value to the list ,so should update 
                  if(headers.containsKey(name)) {
                        headers.get(name).add(value);
                    }else{
                     valueList.add(value);
                    headers.put(name, valueList);
                   }

I saw the javadoc of request.getHeader(name)
     Returns the value of the specified request header
     as a <code>String</code>. If the request did not include a header
     of the specified name, this method returns <code>null</code>.
     ## If there are multiple headers with the same name, this method
     returns the first head in the request.
     The header name is case insensitive. You can use
     this method with any request header.

if you want a list value of special key way don't use 
request.getHeaders(String name);
the doc of this method show below 

    /**
     *
     * Returns all the values of the specified request header
     * as an <code>Enumeration</code> of <code>String</code> objects.
     *
     * <p>Some headers, such as <code>Accept-Language</code> can be sent
     * by clients as several headers each with a different value rather than
     * sending the header as a comma separated list.
     *
     * <p>If the request did not include any headers
     * of the specified name, this method returns an empty
     * <code>Enumeration</code>.
     * The header name is case insensitive. You can use
     * this method with any request header.
     *
     * @param name		a <code>String</code> specifying the
     *				header name
     *
     * @return			an <code>Enumeration</code> containing
     *                  	the values of the requested header. If
     *                  	the request does not have any headers of
     *                  	that name return an empty
     *                  	enumeration. If 
     *                  	the container does not allow access to
     *                  	header information, return null
     *
     */			

    public Enumeration getHeaders(String name);

**Comment by artgon** [comment]
Yeah, that looks like a bug. Please submit a PR and I'll merge it. 

Btw, if you're starting a new project please use the `2.1` branch. The header functionality has been completely rebuilt.

**Comment by xiaoyangW** [comment]
thanks, solved my problem
